controller_interface/test/test_chainable_controller_interface.cpp

```diff
@@ -31,6 +31,26 @@ TEST_F(ChainableControllerInterfaceTest, default_returns)
   EXPECT_FALSE(controller.is_in_chained_mode());
 }
 
+TEST_F(ChainableControllerInterfaceTest, export_internal_state_interfaces)
+{
+  TestableChainableControllerInterface controller;
+
+  // initialize, create node
+  const auto node_options = controller.define_custom_node_options();
+  ASSERT_EQ(
+    controller.init(TEST_CONTROLLER_NAME, "", 50.0, "", node_options),
+    controller_interface::return_type::OK);
+  ASSERT_NO_THROW(controller.get_node());
+
+  auto internal_state_interfaces = controller.export_internal_state_interfaces();
+
+  ASSERT_EQ(internal_state_interfaces.size(), 1u);
```

Comment:
```suggestion
  ASSERT_THAT(internal_state_interfaces, SizeIs(1));
```
I know I'm nitpicking but let's use the advanced features of gmock. The difference is that you get much more informative prints when something isn't right.

You may need to do `using testing::SizeIs;` at the top if it's not there yet